experiments/imit_benchmark.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# This script finds the the mean and standard error of episode return after
+# training GAIL or AIRL on benchmark tasks.
+#
+# The benchmark tasks are defined in the CSV config file
+# `experiments/imit_benchmark_config.csv`.
+
+CONFIG_CSV="experiments/imit_benchmark_config.csv"
+EXPERT_MODELS_DIR="expert_models"
+TIMESTAMP=$(date --iso-8601=seconds)
+LOG_ROOT="output/imit_benchmark/${TIMESTAMP}"
+extra_configs=""
+extra_options=""
+
+SEEDS="0 1 2"
+
+TEMP=$(getopt -o f -l fast,gail,airl,run_name:,log_root:,file_storage: -- $@)
+if [[ $? != 0 ]]; then exit 1; fi
+eval set -- "$TEMP"
+
+while true; do
+  case "$1" in
+    # Fast mode (debug)
+    -f | --fast)
+      CONFIG_CSV="tests/data/imit_benchmark_config.csv"
+      EXPERT_MODELS_DIR="tests/data"
+      SEEDS="0"
+      extra_configs+="fast "
+      shift
+      ;;
+    --gail)
+      extra_configs+="gail "
+      shift
+      ;;
+    --airl)
+      extra_configs+="airl "
+      shift
+      ;;
+    --run_name)
+      extra_options+="--name $2 "
+      shift 2
+      ;;
+    --log_root)
+      LOG_ROOT="$2"
+      shift 2
+      ;;
+    --file_storage)
+      extra_options+="--file_storage $2 "
+      shift 2
+      ;;
+    --)
```

Comment:
When is this needed? It's a common idiom when you want to take some variable-length positional (non-flag) arguments, but we don't seem to use this?

Reply:
I"m pretty sure that GNU getopt's output denotes the end of its options section with a `--`. This while loop was copied from example code (don't remember source) that seemed like idiomatic usage of GNU getopt.

I don't know how otherwise to break the while loop.

(Note that GNU/Linux `getopt` is different from Posix `getopt` and the builtin bash command `getopts`).

Reply:
This is the canonical way to break out of `getopt`, according to these tutorials (hard to google search for because `getopt` is so overloaded and has POSIX and GNU/Linux versions)
https://dustymabe.com/2013/05/17/easy-getopt-for-a-bash-script/
https://www.codebyamir.com/blog/parse-command-line-arguments-using-getopt